### ISSUE TITLE
Flutter plugin template with cocopods issue

### ISSUE THREAD

**Issue by amadeu01** [flutter/flutter#56157]
## Steps to reproduce 

1. `👉  flutter create --template=plugin test_plugin`
2. `👉  cd test_plugin/ios/`
3. `👉  pod lib lint --no-clean`

You will get 

```
 🦄  ✨  
👉  pod lib lint --no-clean

 -> test_plugin (0.0.1)
    - WARN  | license: Missing license type.
    - WARN  | description: The description is equal to the summary.
    - WARN  | [iOS] keys: Missing primary key for `source` attribute. The acceptable ones are: `git, hg, http, svn`.
    - ERROR | [iOS] xcodebuild: Returned an unsuccessful exit code. You can use `--verbose` for more information.
    - NOTE  | xcodebuild:  note: Using new build system
    - NOTE  | xcodebuild:  note: Building targets in parallel
    - NOTE  | [iOS] xcodebuild:  note: Planning build
    - NOTE  | [iOS] xcodebuild:  note: Constructing build description
    - NOTE  | [iOS] xcodebuild:  warning: Skipping code signing because the target does not have an Info.plist file and one is not being generated automatically. (in target 'App' from project 'App')
    - NOTE  | [iOS] xcodebuild:  note: Execution policy exception registration failed and was skipped: Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted" (in target 'test_plugin' from project 'Pods')
    - NOTE  | [iOS] xcodebuild:  note: Execution policy exception registration failed and was skipped: Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted" (in target 'Pods-App' from project 'Pods')
    - ERROR | [iOS] xcodebuild:  /var/folders/k7/1ptshnq15zn3nz6bgp7sj8h88w6vb6/T/CocoaPods-Lint-20200501-27570-50eadt-test_plugin/App/main.swift:1:8: error: could not find module 'test_plugin' for target 'i386-apple-ios-simulator'; found: x86_64-apple-ios-simulator, x86_64

Pods workspace available at `/var/folders/k7/1ptshnq15zn3nz6bgp7sj8h88w6vb6/T/CocoaPods-Lint-20200501-27570-50eadt-test_plugin/App.xcworkspace` for inspection.

[!] test_plugin did not pass validation, due to 2 errors and 3 warnings.
```

Shouldn't the template be the minimum amount of working code to prepare for new plugin development?

**Comment by iapicca** [comment]
Hi @amadeu01 
please follow the steps described in the [docs](https://flutter.dev/docs/development/packages-and-plugins/developing-packages#plugin)
if the issue persists please provide your `flutter doctor -v`
and a reproducible minimal code sample.
Thank you

**Comment by VladyslavBondarenko** [comment]
@amadeu01 
I hope you've solved the issue.
Closing as additional information isn't provided.
Could everyone who still has this problem please file a new issue with the exact descriptions what happens, logs and the output of `flutter doctor -v` please.
